server/isomorphic-routing/index.js

```diff
@@ -5,27 +5,35 @@ import i18n from 'lib/mixins/i18n';
 import sections from '../../client/sections';
 import { serverRender } from 'render';
 import { createReduxStore } from 'state';
+import { dispatchSetSection } from '../../client/controller';
 
 export default function( expressApp, getDefaultContext ) {
 	sections.get()
 		.filter( section => section.isomorphic )
 		.forEach( section => {
-			sections.require( section.module )( serverRouter( expressApp, getDefaultContext ) );
+			sections.require( section.module )( serverRouter( expressApp, getDefaultContext, section ) );
 		} );
 }
 
-function serverRouter( expressApp, mapExpressToPageContext ) {
+function serverRouter( expressApp, mapExpressToPageContext, section ) {
 	return function( route, ...middlewares ) {
-		expressApp.get( route, combinedMiddlewares( mapExpressToPageContext, middlewares ) );
+		expressApp.get( route, combinedMiddlewares( mapExpressToPageContext, middlewares, section ) );
 	}
 }
 
-function combinedMiddlewares( mapExpressToPageContext, middlewares ) {
+function combinedMiddlewares( mapExpressToPageContext, middlewares, section ) {
+	function dispatchSetSectionMiddleware( context, next ) {
+		dispatchSetSection( context, section );
```

Comment:
While perfectly descriptive, these function names (`dispatchSetSection`/`dispatchSetSectionMiddleware`) kinda suck :-) Any alternatives you can think of?